gtsfm/data_association/point3d_initializer.py

```diff
@@ -55,6 +56,54 @@ class TriangulationParam(Enum):
     RANSAC_TOPK_BASELINES = 3  # deterministically choose hypotheses with largest estimate baseline
 
 
+class TriangulationOptions(NamedTuple):
+    """Options for triangulation solver.
+
+    Refs:
+      - https://github.com/colmap/colmap/blob/dev/src/optim/ransac.h
+      - http://www.cse.psu.edu/~rtc12/CSE486/lecture15.pdf
+
+    Args:
+        reproj_error_threshold: the maximum reprojection error allowed.
+        mode: triangulation mode, which dictates whether or not to use robust estimation.
+        min_inlier_ratio: a priori assumed minimum probability that a point is an inlier.
+        confidence: desired confidence that at least one hypothesis is outlier free.
+        dyn_num_hypotheses_multiplier: multiplication factor for dynamically computed hyptheses based on confidence.
+        min_num_hypotheses: minimum number of hypotheses.
+        mas_num_hypotheses: masimum number of hypotheses.
+    """
+
+    reproj_error_threshold: float
+    mode: TriangulationParam
+
+    # RANSAC parameters
+    min_inlier_ratio: float = 0.1
```

Comment:
isnt that a really small ratio? do we have a min_num_inliers too? just to be sure this does not select tracks with one or two inliers?

Reply:
Yes, a smaller ratio will result in a larger number of dynamic hypotheses.

Keep in mind that the measurements are the cameras (with the corresponding keypoint), so a track with one inlier would result in a underdetermined problem (which we check for and triangulation would fail anyways). Two inliers is allowed, but tracks are discarded according to `min_track_length`.

Reply:
Okay, I wonder if a higher min inlier ratio produces better hypothesis? It would definitely save some compute.